### ISSUE TITLE
GNN

### ISSUE THREAD

**Issue by mecsabb** [mecsabb/arbitrage#23]
GNN : G -> (**p**, **v**) where **p** is policy vector and **v** is value vector representing reward of each action compared to random action

G will be a weighted (negatives allowed), directed, cyclic graph with node values the encode the "game state".  Some quick ideas for node values (to be tuned / tested by GNN team):
- make first node -1, visited nodes 1, current node another value, rest of the nodes 0?
- tbd

Naturally, the learner part of the training algorithm is included.  There is a well defined loss function in the paper.

This team will also get task of setting up an environment to compare two models and evaluate which one is better (e.g. generate random graphs and compare mean performance).  We should aim to make this fairly quick as it will be part of the training algorithm.

Things to check out:
some intuition - https://distill.pub/2021/gnn-intro/
a helpful framework - https://pytorch-geometric.readthedocs.io/en/latest/get_started/colabs.html
